msgp/errors_tinygo.go

```diff
@@ -0,0 +1,43 @@
+// +build tinygo
+
+package msgp
+
+import (
+	"reflect"
+)
+
+// ctxString converts the incoming interface{} slice into a single string,
+// without using fmt under tinygo
+func ctxString(ctx []interface{}) string {
+	out := ""
+	for idx, cv := range ctx {
+		if idx > 0 {
+			out += "/"
+		}
+		out += ifToStr(cv)
+	}
+	return out
+}
+
+type stringer interface {
+	String() string
+}
+
+func ifToStr(i interface{}) string {
+
+	switch v := i.(type) {
+	case stringer:
+		return v.String()
+	case error:
+		return v.Error()
+	case string:
+		return v
+	default:
+		return reflect.ValueOf(i).String()
+	}
+
+}
+
+func quoteStr(s string) string {
+	return `"` + s + `"`
```

Comment:
This doesn't do quite the same thing as `%q` for control characters. Since `strconv` is already a dependency of the `tinygo` build, can we use `strconv.Quote()` here?

Reply:
see notes on strconv.Quote code size